### DISCUSSION TITLE
Minified React error #418 in production

### DISCUSSION THREAD

**Discussion by MasterProvider** [vercel/next.js#74305] [Help]
### Summary

In development
![Screenshot 2024-12-26 142717](https://github.com/user-attachments/assets/e6e68284-ca23-40ea-b8ad-488ce4f1ba6b)

In production
![Screenshot 2024-12-26 142655](https://github.com/user-attachments/assets/725be5ee-f0c9-4acb-8285-10aab3f5ba75)

Is this fine? It's so annoying to have hydration errors in production, especially when there are no errors in development.
This occurred only in console log

### Additional information

```js
"dependencies": {
    "@emotion/react": "^11.14.0",
    "@emotion/styled": "^11.14.0",
    "next": "^15.1.2",
    "react": "^19.0.0",
    "react-dom": "^19.0.0",
     ...
  },
  "devDependencies": {
    "@eslint/eslintrc": "^3",
    "eslint": "^9",
    "eslint-config-next": "^15.1.2"
  }
```

**Comment by MasterProvider** [comment]
Is this problem in Next.js, or is it in React 19 itself?

**Comment by icyJoseph** [comment]
These are React hydration issues. Any party that manipulates JSX to produce the first server HTML, and even those that pre-process it in the browser can be at fault.

With that being said, there is a new experimental, Next.js config, `experimental.allowDevelopmentBuild`, to help you debug this kind of issue, https://github.com/vercel/next.js/pull/65463

So you'd need to add that to your next.config.js experimental:
```js
const config: NextConfig = {
  experimental: {
      allowDevelopmentBuild: true
  }
};

```

And then run `NODE_ENV=development npm run build`, on your terminal, does it give you more information?

**Reply by icyJoseph** [reply]
Another reason to see this kind of issue, is code that uses certain conditionals, such as timestamps and such, or takes different codepaths during dev and prod builds...

**Comment by MasterProvider** [comment]
This issue was resolved by not initializing real-time timestamps.
